src/main/scala/com/bazaarvoice/sswf/model/ScheduledStep.scala

```diff
@@ -13,4 +17,9 @@ case class DefinedStep[StepEnum <: (Enum[StepEnum] with WorkflowStep)](step: Ste
   stepInput.stepInputString.foreach(input => require(!input.contains("\u0000"), "input may not contain a null byte."))
 
   def withoutResume = this.copy(stepInput = stepInput.copy(resumeProgress = None))
+
+  override def isSameStepAs(other: ScheduledStep[StepEnum]): Boolean = other match {
```

Comment:
also, remember that overriding equals also means you have to override hashCode, which means you're changing the way set and map collections will treat this object. To do that safely, you need to know all the ways the object is now and will ever be used.

In other words, it's far less impact to define your own equals-like method rather than trying to re-use existing semantics.

There is a similar lesson with toString. If you read the Java spec, they say that toString is for developer readability _only_. Any other use is non-standard and potentially dangerous. The danger comes from the same place as tweaking equals or hashcode. Future developers (including you) acting in compliance with the Java spec may reasonably change toString to produce any string they think is more readable at the time. If you're also using it for serialization, the code will break unexpectedly.